wordpress-mobile/WordPress-Android
# Delete cancelled media from media library

Fixes #5942, partially. When manually cancelling media uploads from the editor, they'll now be deleted, and won't show up as permanently 'Uploading' in the media browser:

![eternal-media-upload](https://cloud.githubusercontent.com/assets/9613966/26459051/308d7c12-4143-11e7-8754-5dd96867fc66.png)

If the user exits the editor with in-progress media, they won't be deleted (as the user might re-open the post and retry the media). Instead, they'll be marked as failed, and will show up with a 'Retry' message in the media browser:

![retry-media-library](https://cloud.githubusercontent.com/assets/9613966/26790278/219ab55c-49e1-11e7-98a3-4497067411ff.png)

This still isn't ideal. Retrying from the media browser will work, and the media will be uploaded to the site, but the post won't be updated to use the remote media URL. This will be fixed once the async media branch is in `develop`, and in my opinion this is as far as it's worth fixing `develop` in the meantime.

**Note:** ~~https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/457 should be reviewed in parallel, and this PR should be updated to use the latest FluxC `develop` hash before being merged.~~ https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/457 merged and hash updated in 4f51a5c.

Note: A separate PR with a more satisfying fix will be made against the `feature/async-media` branch.

To test:
1. Upload media to the editor
2. Cancel the media
3. Verify that the media doesn't exist in the media library
4. Upload another media to the editor
5. Exit the editor while media is in progress
6. Verify that the media shows up with a 'Retry' message in the media library

Try the above for WP.com and self-hosted sites.

cc @mzorz
